crates/tests/mqtt_tests/src/test_mqtt_server.rs

```diff
@@ -115,7 +114,11 @@ fn spawn_broker() -> u16 {
         }
 
         match broker_thread.join() {
-            Ok(Ok(())) => unreachable!("`broker.start()` does not terminate"),
+            Ok(Ok(())) => {
+                // I don't know why it happened, but I have observed this once while testing
+                // So just log the error and retry starting the broker on a new port
+                eprintln!("MQTT-TEST ERROR: `broker.start()` should not terminate until after `spawn_broker` returns")
```

Comment:
But when this happens and when we retry the broker start on the next iteration, the client thread that's started (line. 137) in the last iteration also must be aborted somehow, right? Because once that client thread enters the loop, I don't see how it can break out even on a connection error because of the very specific `if let` check.

Reply:
The client thread is not part of the loop, it's only started once we have a healthy broker.